### DISCUSSION TITLE
L.DoFile does not work when i require a so in lua script

### DISCUSSION THREAD

**Discussion by littleningmeng** [yuin/gopher-lua#443] [Q&A]
I have a lua script a.lua, and in this file I required a Lua so module:  
```
pb = require "pb"
print(pb)
```
and the `pb` module is a lua so in the same dir with a.lua
but, when the Golang code below not work  
```
L := lua.NewState()
defer L.Close()
if err := L.DoFile("a.lua"); err != nil {
  panic(err)
}
```

error message:  
```
panic: <string>:1: module pb not found:
	no field package.preload['pb']
	stat ./pb.lua: no such file or directory
	stat /usr/local/share/lua/5.1/pb.lua: no such file or directory
	stat /usr/local/share/lua/5.1/pb/init.lua: no such file or directory, 
```

**Comment by cloudwindy** [comment]
Where are you calling your Go program? Your working directory decides where to find that library.

**Reply by littleningmeng** [reply]
Actually so-loader is not supported in gopher-lua.
I wrote a C wrapper for solving this problem and it works.

**Comment by cloudwindy** [comment]
If you do want to require scripts from relative directory (where the script is), you can try [this](https://stackoverflow.com/questions/9145432/load-lua-files-by-relative-path)

**Comment by cloudwindy** [comment]
> Actually so-loader is not supported in gopher-lua. I wrote a C wrapper for solving this problem and it works.

Oh, didn't see you're trying to load a .so file! Glad you have solved it.
